node/node.go

```diff
@@ -509,7 +492,7 @@ func (node *Node) Start(ctx context.Context) error {
 		return errors.Wrap(err, "Could not make new storage client")
 	}
 
-	node.RetrievalClient = retrieval.NewClient(node)
+	node.setupProtocols()
```

Comment:
personally i like having all the construction inline so i can see what gets what, but to each her own

Reply:
Start() is hella long though. I want to abstract more stuff.

Reply:
it definitely needs abstracting. partitioning helps i guess but we need a builder thingy to really help

Reply:
Chiming in: I prefer your way @shannonwells